web_client/src/App.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, Fragment } from 'react';
+import { Route } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { ConnectedRouter } from 'connected-react-router';
+import { handleApplicationStarted } from 'actions/app';
+import { getLoading } from 'selectors';
+import Loading from 'scenes/Loading';
+import Landing from 'scenes/Landing';
+import Playlists from 'scenes/Playlists';
+import SouffleStation from 'scenes/SouffleStation';
+
+const App = ({ history, loading }) => (
+  loading ? <Loading /> :
```

Comment:
you don't want loading to be it's own route within connectedrouter? or a component that other routes use?

Reply:
> you don't want loading to be it's own route within connectedrouter?

I don't want the `loading` state to change the actual browser route.

I think it'd look jumpy / confusing for a user to see their browser path say '/loading' and then change back. 

unless there's a way to pick a route based on state rather than path? i was a little confused about this

> or a component that other routes use?

given that it's a pretty small application i don't mind the app either rendering the loading page or, if the data's been loaded, whatever route the user is at.

Reply:
yeah then i agree that `withLoading` HOC is probably a good idea.